controllers/adminsController/adminDeleteComment.js

```diff
@@ -0,0 +1,61 @@
+const Comments = require("../../models/comments");
+const Applications = require("../../models/applications");
+const { ObjectId } = require("mongoose").Types;
+
+// Utilities
+const CustomError = require("../../utils/customError");
+const responseHandler = require("../../utils/responseHandler");
+const commentHandler = require("../../utils/commentHandler");
+
+/**
+ * @author David Okanlawon
+ *
+ * Deletes a single comment.
+ *
+ * @param {*} req - The request object
+ * @param {*} res - The response object
+ * @param {*} next - The function executed to call the next middleware
+ */
+const adminDeleteComment = async (req, res, next) => {
+  const commentId = req.body.commentId;
+  const { organizationId } = req.token;
+
+  if (!ObjectId.isValid(organizationId))
+    return next(new CustomError(401, "unauthorised request"));
```

Comment:
Given that you are checking if the id is valid or not and not if its authorized, shouldn't you be returning a 422?

Reply:
Given that this is not a visible parameter but a value encoded in token payload, this is an authentication error. sending a 422 will be misleading. This tells that something is wrong with their token not a value provided. Validation errors are for input values

Reply:
Makes sense.